components/GameDetailsView.jsx

```diff
@@ -0,0 +1,217 @@
+import { useRouter } from "next/router";
+import { useState } from "react";
+import BackButton from "@/components/buttons/BackButton";
+import styled from "styled-components";
+import {
+  StyledButton,
+  StyledLinkButton,
+  StyledButtonWrapper,
+  StyledDeleteButton,
+} from "@/components/buttons/DefaultButtons";
+
+export default function GameDetailsView({
+  selectedGame,
+  onEditGame,
+  onDeleteGame,
+  gameId,
+}) {
+  const router = useRouter();
+  const [buttonMode, setButtonMode] = useState("default");
+  const [sliderValue, setSliderValue] = useState(selectedGame.progress);
+
+  function handleChange(event) {
+    selectedGame = { ...selectedGame, status: event.target.value };
+    onEditGame(selectedGame);
+  }
+
+  function handleProgressChange(event) {
+    const newValue = parseInt(event.target.value, 10);
+    setSliderValue(newValue);
+    if (newValue === 100) {
+      selectedGame = { ...selectedGame, status: "Completed", progress: 100 };
+    } else if (newValue === 0) {
+      selectedGame = { ...selectedGame, status: "Planned", progress: 0 };
+    } else {
+      selectedGame = {
+        ...selectedGame,
+        status: "In Progress",
+        progress: newValue,
+      };
+    }
+    onEditGame(selectedGame);
```

Comment:
I would make a small change to these lines of code, since what we are currently doing is overwriting the `selectedGame` prop, which is possible, but not a good pattern. Instead I would create a new variable that is based on `selectedGame`:

```suggestion
    let updatedGame = {};
    if (newValue === 100) {
      updatedGame = { ...selectedGame, status: "Completed", progress: 100 };
    } else if (newValue === 0) {
      updatedGame = { ...selectedGame, status: "Planned", progress: 0 };
    } else {
      updatedGame = {
        ...selectedGame,
        status: "In Progress",
        progress: newValue,
      };
    }
    onEditGame(updatedGame);
```